core/src/main/scala/kafka/log/LogConfig.scala

```diff
@@ -115,67 +117,214 @@ object LogConfig {
+    "durability and allow the operating system's background flush capabilities as it " +
+    "is more efficient. This setting can be overridden on a per-topic basis (see <a " +
+    "href=\"#topic-config\">the per-topic configuration section</a>)."
+  val FlushMsDoc = "This setting allows specifying a time interval at which we will " +
+    "force an fsync of data written to the log. For example if this was set to 1000 " +
+    "we would fsync after 1000 ms had passed. In general we recommend you not set " +
+    "this and use replication for durability and allow the operating system's background " +
+    "flush capabilities as it is more efficient."
+  val RetentionSizeDoc = "This configuration controls the maximum size a log can grow " +
+    "to before we will discard old log segments to free up space if we are using the " +
+    "\"delete\" retention policy. By default there is no size limit only a time limit."
+  val RetentionMsDoc = "This configuration controls the maximum time we will retain a " +
+    "log before we will discard old log segments to free up space if we are using the " +
+    "\"delete\" retention policy. This represents an SLA on how soon consumers must read " +
+    "their data."
+  val MaxIndexSizeDoc = "This configuration controls the size of the index that maps " +
+    "offsets to file positions. We preallocate this index file and shrink it only after log " +
+    "rolls. You generally should not need to change this setting."
+  val MaxMessageSizeDoc = "This is largest message size Kafka will allow to be appended. Note that if you increase" +
+    " this size you must also increase your consumer's fetch size so they can fetch messages this large."
+  val IndexIntervalDoc = "This setting controls how frequently Kafka adds an index " +
+    "entry to it's offset index. The default setting ensures that we index a message " +
+    "roughly every 4096 bytes. More indexing allows reads to jump closer to the exact " +
+    "position in the log but makes the index larger. You probably don't need to change " +
+    "this."
   val FileDeleteDelayMsDoc = "The time to wait before deleting a file from the filesystem"
-  val DeleteRetentionMsDoc = "The time to retain delete markers in the log. Only applicable for logs that are being" +
-    " compacted."
-  val MinCleanableRatioDoc = "The ratio of bytes that are available for cleaning to the bytes already cleaned"
-  val CompactDoc = "Should old segments in this log be deleted or deduplicated?"
-  val UncleanLeaderElectionEnableDoc = "Indicates whether unclean leader election is enabled"
-  val MinInSyncReplicasDoc = "If number of insync replicas drops below this number, we stop accepting writes with" +
-    " -1 (or all) required acks"
+  val DeleteRetentionMsDoc = "The amount of time to retain delete tombstone markers " +
+    "for <a href=\"#compaction\">log compacted</a> topics. This setting also gives a bound " +
```

Comment:
I'm not sure either.

On the one hand, it makes the docs clearer and better. It's really useful to click from the config section (which contains a simple one-line description) to the relevant section of the doc (which explains stuff in much more detail).

On the other hand, I agree that it's weird to have references in it that are not useful in isolation. The links will look weird and not make sense in the toRst() output, for example.

We could add the idea of a "footnote", which is added to the docs to say "for more info, go to see this section".  It would be something on the ConfigDef, and the toHtmlString() method could use this footnote to append to the description, and the toRst() method would not. Hm, maybe it should be called an "HTML footnote"?

Not sure if it's worth building that code infrastructure.